src/internal/SignalGenerator.h

```diff
@@ -0,0 +1,20 @@
+#ifndef _INTERNAL_SIGNAL_GENARATOR_HPP_
+#define _INTERNAL_SIGNAL_GENARATOR_HPP_
+
+#include <math.h>
+
+class SignalGenerator{
+
+  double m_deg;  // Waveform frequency
+  double m_amp;  // Waveform amplitude
+  double m_step; // Sampling step
+
+  public:
+  SignalGenerator (void);
+  int setAmp (double amp);
```

Comment:
I would create `getters` functions here. You can implement getters and setters function inline in the header file.

Reply:
I left it the way it was for simplicity. Next time I will follow the advice.